include/sampling/sampling.hpp

```diff
@@ -68,169 +60,157 @@ template <
-    typedef RandomPointGenerator<walk> RandomPointGenerator;
+    typename WalkTypePolicy::template Walk<Polytope, RandomNumberGenerator> walk(P, starting_point, rng);
 
-    Point p = starting_point;
+    Point p = starting_point; 
     if (nburns > 0) {
-        RandomPointGenerator::apply(P, p, nburns, walk_len, randPoints,
-                                    push_back_policy, rng, WalkType.param);
-        randPoints.clear();
+        for (unsigned int i = 0; i < nburns; ++i) {
+            walk.apply(P, p, walk_len, rng);  
+        }
+    }
+
+    randPoints.clear();
+
+    for (unsigned int i = 0; i < rnum; ++i) {
+        walk.apply(P, p, walk_len, rng);  
+        randPoints.push_back(p);  
     }
-    RandomPointGenerator::apply(P, p, rnum, walk_len, randPoints,
-                                push_back_policy, rng, WalkType.param);
 }
 
 
-template
-<
-        typename WalkTypePolicy,
-        typename PointList,
-        typename Polytope,
-        typename RandomNumberGenerator,
-        typename Point
+
+template <
+    typename WalkTypePolicy,
+    typename PointList,
+    typename Polytope,
+    typename RandomNumberGenerator,
+    typename Point
 >
 void uniform_sampling_boundary(PointList &randPoints,
-                            Polytope &P,
-                            RandomNumberGenerator &rng,
-                            const unsigned int &walk_len,
-                            const unsigned int &rnum,
-                            const Point &starting_point,
-                            unsigned int const& nburns)
-{
-    typedef typename WalkTypePolicy::template Walk
-            <
-                    Polytope,
-                    RandomNumberGenerator
-            > walk;
+                               Polytope &P,
+                               RandomNumberGenerator &rng,
+                               const unsigned int &walk_len,
+                               const unsigned int &rnum,
+                               Point &starting_point, 
+                               unsigned int const& nburns) {
 
-    //RandomNumberGenerator rng(P.dimension());
-    PushBackWalkPolicy push_back_policy;
+    
+    typename WalkTypePolicy::template Walk<Polytope, RandomNumberGenerator> walk(P, starting_point, rng);
 
-    Point p = starting_point;
+    Point p1 = starting_point; 
 
-    typedef BoundaryRandomPointGenerator <walk> BoundaryRandomPointGenerator;
+    Point p2 = starting_point; 
     if (nburns > 0) {
-        BoundaryRandomPointGenerator::apply(P, p, nburns, walk_len,
-                                            randPoints, push_back_policy, rng);
-        randPoints.clear();
+        for (unsigned int i = 0; i < nburns; ++i){
+            walk.apply(P, p1, p2, walk_len, rng);
+        }
     }
-    unsigned int n = rnum / 2;
-    BoundaryRandomPointGenerator::apply(P, p, rnum / 2, walk_len,
-                                        randPoints, push_back_policy, rng);
 
+    
+    randPoints.clear();
```

Comment:
is this needed?

Reply:
similar in all walks

Reply:
Initially, I did not have this function. However, I thought that it might be a good practice to ensure that everything is cleared before performing the actual sampling. Besides, this operation should be done in constant time (correct me if I am wrong), especially in the case in which we expect randPoints to be empty. Consequently, I thought that there would be no problem in adding it just for safety.